streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java

```diff
@@ -203,6 +208,34 @@ public void shouldBeAbleToRunWithTwoSubtopologiesAndMultiplePartitions() throws
         runSimpleCopyTest(1, MULTI_PARTITION_INPUT_TOPIC, MULTI_PARTITION_THROUGH_TOPIC, MULTI_PARTITION_OUTPUT_TOPIC, false, eosConfig);
     }
 
+    // This is technically a purely producer-client test, but since we're relying on the specific error message being
+    // thrown we should make sure it can't change without us noticing. Once KAFKA-10186 is resolved we should fix this
+    @Test
+    public void testExceptionForPendingUnflushedDataWhenTransactionIsAborted()  {
+        final Map<String, Object> configs = new HashMap<>();
+        configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        configs.put("client.id", "client-1");
+        configs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, "txnId");
+
+        final KafkaProducer<String, String> producer =
+            new KafkaProducer<>(configs, new StringSerializer(), new StringSerializer());
+
+        final ProducerRecord<String, String> record = new ProducerRecord<>(SINGLE_PARTITION_INPUT_TOPIC, "value");
+
+        producer.initTransactions();
+        producer.beginTransaction();
+
+        producer.send(record);
+
+        final AtomicReference<Exception> receivedException = new AtomicReference<>(null);
+        producer.send(record, (recordMetadata, exception) -> receivedException.compareAndSet(null, exception));
```

Comment:
Do we need to make them producer config changes to ensure that the producer does not flush the record by chance? (eg, increase `linger.ms` to `MAX_VALUE` or similar?)

Reply:
Hm, yeah, good call. I didn't realize this defaults to 0